### PR TITLE
Feature/project details

### PR DESCRIPTION
**Pull request by eduskx**
please have a look :)

### CODE REVIEW

File: pages/projects/[id]/index.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import Image from "next/image";
+import { FaArrowLeftLong } from "react-icons/fa6";
+import { uid } from "uid";
```
Comment (Roland-Hufnagel) [inline pages/projects/[id]/index.js:5]: You don't use it anywhere so please remove it.

File: pages/projects/[id]/index.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import styled from "styled-components";
+
+export default function ProjectDetailsPage({ projects }) {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const projectData = projects.find((project) => project.id === id);
+
+  if (!projectData) {
+    return null;
```
Comment (Roland-Hufnagel) [inline pages/projects/[id]/index.js:15]:
```suggestion
    return <h1>No project found<h1>;

```
This is more suitable than a blank site

File: pages/projects/[id]/index.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  } = projectData;
+
+  return (
+    <>
+      <Link href="/">
+        <FaArrowLeftLong /> Back
+      </Link>
+
+      <StyledDetailsWrapper>
+        <h2>{title}</h2>
```
Comment (Roland-Hufnagel) [inline pages/projects/[id]/index.js:35]: You should make sure to have a `<h1>` before using a `<h2>`.

File: pages/projects/[id]/index.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        <StyledImageWrapper>
+          <Image src={imageUrl} alt={title} width={300} height={200} />
+          <StyledP>{complexity}</StyledP>
+        </StyledImageWrapper>
+        <p>{description}</p>
+        <StyledDuration>Duration: {duration}</StyledDuration>
+
+        <StyledMaterialsList>
+          <h3>Materials</h3>
```
Comment (Roland-Hufnagel) [inline pages/projects/[id]/index.js:44]: A list should only have list-items as direct children. You can place the headline above the list.

File: pages/projects/[id]/index.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+          {materials.map((material) => (
+            <StyledListItems key={uid()}>{material}</StyledListItems>
+          ))}
+        </StyledMaterialsList>
+
+        <StyledInstructionsList>
+          <h3>Instructions</h3>
```
Comment (Roland-Hufnagel) [inline pages/projects/[id]/index.js:51]: same as above.

File: pages/projects/[id]/index.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    <>
+      <Link href="/">
+        <FaArrowLeftLong /> Back
+      </Link>
+
+      <StyledDetailsWrapper>
+        <h2>{title}</h2>
+        <StyledImageWrapper>
+          <Image src={imageUrl} alt={title} width={300} height={200} />
+          <StyledP>{complexity}</StyledP>
```
Comment (Roland-Hufnagel) [inline pages/projects/[id]/index.js:38]: Please do not use such abbreviations (`<StyledP>`) . This makes code hard to read.

File: pages/projects/[id]/index.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        </StyledImageWrapper>
+        <p>{description}</p>
+        <StyledDuration>Duration: {duration}</StyledDuration>
+
+        <StyledMaterialsList>
+          <h3>Materials</h3>
+          {materials.map((material) => (
+            <StyledListItems key={uid()}>{material}</StyledListItems>
```
Comment (Roland-Hufnagel) [inline pages/projects/[id]/index.js:46]: To make use of reacts power you should not create new uids as a key. A better solution is `key={material}` if you are sure that each material exists only one time and the best solution is to make sure each material has an id that can be used as a key.

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Hey team. This looks already really good 💪. I left some comments to improve your code. Please have a look. 
One point apart from this code review: I recommend to start from the beginning with a responsive website (mobile first). If you make these adjustments right from the beginning it could save a lot of work later (but it's up to you... 😀)

File: pages/projects/[id]/index.js (hunk 1 above)
Comment (eduskx) [reply pages/projects/[id]/index.js:5]: We aren't sure what exactly you mean because we checked and we actually use all of these

**Review by MerleMak — APPROVED** [review]
nice!

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Everything is fine now except the key prop. Please have a look.

File: pages/projects/[id]/index.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+          <Image src={imageUrl} alt={title} width={300} height={200} />
+          <StyledComplexityTag>{complexity}</StyledComplexityTag>
+        </StyledImageWrapper>
+        <p>{description}</p>
+        <StyledDuration>Duration: {duration}</StyledDuration>
+
+        <h2>Materials</h2>
+        <StyledMaterialsList>
+          {materials.map((material) => (
+            <StyledListItems key={material.index}>{material}</StyledListItems>
```
[the remark this answers is not in the document]
Comment (eduskx) [reply pages/projects/[id]/index.js:45]: changed it, thank you!

{materials.map((material, index) => (
            <StyledListItems key={index}>{material}</StyledListItems>
          ))}

**Review by Roland-Hufnagel — APPROVED** [review]
LGTM

**Pull request merged**